src/game/SpellEffects.cpp

```diff
@@ -2850,115 +2844,152 @@ void Spell::EffectTameCreature(SpellEffectIndex /*eff_idx*/)
-            {
-                ((Player*)m_caster)->PetSpellInitialize();
-            }
-            return;
-        }
+                        m_caster->GetMap()->Add((Creature*)OldSummon);
 
-        if (m_caster->GetTypeId() == TYPEID_PLAYER)
-            OldSummon->Unsummon(OldSummon->getPetType() == HUNTER_PET ? PET_SAVE_AS_DELETED : PET_SAVE_NOT_IN_SLOT, m_caster);
-        else
-            return;
-    }
+                        ((Player*)m_caster)->PetSpellInitialize();
+                    }
 
-    CreatureInfo const* cInfo = petentry ? ObjectMgr::GetCreatureTemplate(petentry) : nullptr;
+                    delete NewSummon;
+                }
 
-    // == 0 in case call current pet, check only real summon case
-    if (petentry && !cInfo)
-    {
-        sLog.outErrorDb("EffectSummonPet: creature entry %u not found for spell %u.", petentry, m_spellInfo->Id);
-        return;
-    }
+                // Load pet from db; if we have anything to load
+                else if (NewSummon->LoadPetFromDB((Player*)m_caster, petentry))
+                    delete OldSummon;
 
-    Pet* NewSummon = new Pet;
+                else
+                {
+                    delete OldSummon;
+                    delete NewSummon;
+                    // TODO: Inform the hunter he has no pet
+                }
 
-    // petentry==0 for hunter "call pet" (current pet summoned if any)
-    if (m_caster->GetTypeId() == TYPEID_PLAYER && NewSummon->LoadPetFromDB((Player*)m_caster, petentry))
-    {
-        if (NewSummon->getPetType() == SUMMON_PET)
-        {
-            // Remove Demonic Sacrifice auras (known pet)
-            Unit::AuraList const& auraClassScripts = m_caster->GetAurasByType(SPELL_AURA_OVERRIDE_CLASS_SCRIPTS);
-            for (Unit::AuraList::const_iterator itr = auraClassScripts.begin(); itr != auraClassScripts.end();)
+                return;
+            }
+            case CLASS_WARLOCK:
             {
-                if ((*itr)->GetModifier()->m_miscvalue == 2228)
+                // Remove Demonic Sacrifice auras (known pet)
+                Unit::AuraList const& auraClassScripts = m_caster->GetAurasByType(SPELL_AURA_OVERRIDE_CLASS_SCRIPTS);
+                for (Unit::AuraList::const_iterator itr = auraClassScripts.begin(); itr != auraClassScripts.end();)
                 {
-                    m_caster->RemoveAurasDueToSpell((*itr)->GetId());
-                    itr = auraClassScripts.begin();
+                    if ((*itr)->GetModifier()->m_miscvalue == 2228)
+                    {
+                        m_caster->RemoveAurasDueToSpell((*itr)->GetId());
+                        itr = auraClassScripts.begin();
+                    }
+                    else
+                        ++itr;
                 }
+
+                if (OldSummon)
+                    OldSummon->Unsummon(PET_SAVE_NOT_IN_SLOT, m_caster);
+
+                else
```

Comment:
I'm waiting for the checks to finish, when they are done and show that it's all good, I'm gonna hotfix the else statement away and then merge with master.
Sounds good to you guys?